backend/core/src/main/java/org/sonarsource/sonarlint/core/repository/connection/AbstractConnectionConfiguration.java

```diff
@@ -75,6 +76,14 @@ public boolean isSameServerUrl(String otherUrl) {
     return Objects.equals(myUri, otherUri);
   }
 
+  public void setHasEnterpriseCSharpPlugin(boolean hasEnterpriseCSharpPlugin) {
+    this.hasEnterpriseCSharpPlugin = hasEnterpriseCSharpPlugin;
+  }
+
+  public boolean hasEnterpriseCSharpPlugin() {
+    return hasEnterpriseCSharpPlugin;
+  }
```

Comment:
Related to my previous comment, and maybe it's more of a personal taste. Having setters/getters makes it usually harder to follow the flow: when is it called? Are we calling the setter in all possible situations where it needs to be called? I have been trying recently to avoid setters if possible, and prefer immutable objects